lib/xlsx_reader/parsers/worksheet_parser.ex

```diff
@@ -22,6 +22,8 @@ defmodule XlsxReader.Parsers.WorksheetParser do
               cell_style: nil,
               value: nil,
               formula: nil,
+              shared_formula_string_index: nil,
+              shared_formulas: %{},
```

Comment:
We could use the `XlsxReader.Array` module to store the indexed formulas using the OTP `:array` module which is significantly faster for lookups. The index will have to be converted to an integer though.

Reply:
Good idea! Done.